packages/dapp/src/components/Widgets/Widgets.tsx

```diff
@@ -26,41 +26,44 @@ export function Widgets() {
   const [starterVariantUsed, setStarterVariantUsed] = useState(false);
   const [_starterVariant, setStarterVariant] = useState<
     WidgetSubvariant | 'buy'
-  >(TabsMap.Exchange.value);
+  >(TabsMap.Exchange.variant);
 
   const starterVariant: StarterVariantType = useMemo(() => {
     let url = window.location.pathname.slice(1);
     if (Object.values(LinkMap).includes(url as LinkMap)) {
-      if (url === TabsMap.Exchange.value) {
-        return TabsMap.Exchange.value;
-      } else if (url === TabsMap.Refuel.value) {
-        return TabsMap.Refuel.value;
+      if (!!TabsMap.Buy.destination.filter((el) => el === url).length) {
+        return TabsMap.Buy.variant;
+      } else if (
+        !!TabsMap.Refuel.destination.filter((el) => el === url).length
+      ) {
+        return TabsMap.Refuel.variant;
       } else {
-        return TabsMap.Buy.value;
+        return TabsMap.Exchange.variant;
       }
     } else {
-      return TabsMap.Exchange.value;
+      // default and fallback: Exchange-Tab
+      return TabsMap.Exchange.variant;
     }
   }, []);
 
   const getActiveWidget = useCallback(() => {
     if (!starterVariantUsed) {
-      starterVariant === TabsMap.Exchange.value
+      starterVariant === TabsMap.Exchange.variant
```

Comment:
can we convert this to a switch case logic instead of chained ternary?


```
switch(starterVariant) {
    case TabsMap.Exchange.variant:
        setActiveTab(TabsMap.Exchange.index)
    .
    .
    .
}
```